code/modules/mob/living/carbon/xenomorph/castes/Queen.dm

```diff
@@ -517,13 +521,44 @@
 				overwatch(observed_xeno, TRUE)
 
 		if(ovipositor && !is_mob_incapacitated(TRUE))
-			egg_amount += 0.07 //one egg approximately every 30 seconds
+			if(egg_autoplant)
+				egg_amount += 0.0525 // one egg every 45 seconds. Ovi is now planting eggs as well as making them so it's slower.
+			else
+				egg_amount += 0.07 //one egg approximately every 30 seconds
+
 			if(egg_amount >= 1)
-				if(isturf(loc))
-					var/turf/T = loc
-					if(T.contents.len <= 25) //so we don't end up with a million object on that turf.
-						egg_amount--
-						new /obj/item/xeno_egg(loc, hivenumber)
+				egg_amount--
+				if(egg_autoplant)
+					INVOKE_ASYNC(src, PROC_REF(trigger_autoplant))
+				else if(isturf(loc)) //so we don't end up with a million object on that turf.
+					var/turf/turf = loc
+					if(turf.contents.len <= 25)
+						new /obj/item/xeno_egg(loc, hivenumber)  // plop an egg
+
+/mob/living/carbon/xenomorph/queen/proc/trigger_autoplant()
+	// viable turfs within surroundings we can plant eggs on
+	var/list/turf/suitable_turfs = list()
+	// turfs within range and view
+	var/list/turf/surroundings = oview(egg_planting_range, src)
+
+	for(var/turf/turf in surroundings)
+		var/obj/effect/alien/weeds/weeds = locate(/obj/effect/alien/weeds) in turf
+		if(check_alien_construction(turf, silent = TRUE) && weeds)
```

Comment:
There are additional suitability checks not being performed here. E.g. 
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/c7a66501-23f3-42ec-92b9-ec5188db0465)
Message is from me manually trying to place it there, but autoplant places it where I cannot. Be sure to replicate the same checks `/obj/item/xeno_egg/proc/plant_egg` performs, or even better have a proc that both can call to  return whether that location is suitable or not (you'll just want an argument controlling whether it gives to_chat messages or not - because the automatic planting shouldn't be spamming the user with invalid locations - it should just silently check all around it).

Egg planting also normally costs plasma; though its small, I still think this should be retained. This should only be a QoL feature not different in any other way to what you could do normally.